revel/run.go

```diff
@@ -31,41 +32,92 @@ You can set a port as an optional third parameter.  For example:
     revel run github.com/revel/samples/chat prod 8080`,
 }
 
+// RunArgs is hold revel run parameters
+type RunArgs struct {
+	ImportPath string
+	Mode       string
+	Port       int
+}
+
 func init() {
 	cmdRun.Run = runApp
 }
 
-func runApp(args []string) {
-	if len(args) == 0 {
-		errorf("No import path given.\nRun 'revel help run' for usage.\n")
+func parseRunArgs(args []string) *RunArgs {
+	argsLength := len(args)
+	ia := RunArgs{
+		ImportPath: getImportPathCurrentDir(),
+		Mode:       DefaultRunMode,
+		Port:       revel.HTTPPort,
 	}
 
-	// Determine the run mode.
-	mode := DefaultRunMode
-	if len(args) >= 2 {
-		mode = args[1]
+	if argsLength == 3 {
+		port, err := strconv.Atoi(args[2])
+		if err != nil {
+			errorf("Failed to parse port as integer: %s", args[2])
+		}
+		ia.ImportPath = args[0]
+		ia.Mode = args[1]
+		ia.Port = port
+	} else if argsLength == 2 {
+		// args length 2 may have following combination
```

Comment:
"args length 2 may have following combination" to "Possible combinations" to make comment more concise. Same for other case comment blocks.